### PR TITLE
Accept Nans in Normalize 

### PR DESCRIPTION
**Pull request by twennemann**
Solves #389 

We decided
- We allow NaNs in normalize

### CODE REVIEW

**Comment by f-brinkmann** [comment]
Thanks for the effort. The code looks good. But I wonder if we should discuss this in more detail. Which functions should allow NaNs? Should there be a flag, that has to be set or should it be allowed by default? Is there a good way of testing this for all functions that allow it?

**Review by ahms5 — COMMENTED** [review]
looks good to me, I think we should add that nans are ignored in the doc.

**Comment by f-brinkmann** [comment]
We discuss that we explicitly want to add an `ignore_nan` flag to the functions that except time data and NaNs would be a problem. This is also relevant for `dsp.average`.

**Comment by mberz** [comment]
This is an example of the naming convention scipy follows:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pmean.html#scipy.stats.pmean

**Comment by twennemann** [comment]
Added `nan_policy` parameter to normalize function. It follows scipys naming convention. 
Use:
- `propagate` --> Output will be NaN if Signal includes NaNs (with np.max/np.mean)
- `omit` --> NaNs will be omitted in the normalization (with np.nanmax/np.nanmean)
- `raise`--> A `ValueError` will be raised if the signal includes NaNs

I thought `nan_policy='raise'` might be a logical default, so the user will be informed that the signal includes NaNs and to choose a policy.

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -1997,7 +1997,7 @@ def average(signal, mode='linear', caxis=None, weights=None, keepdims=False):
 
 def normalize(signal, reference_method='max', domain='time',
               channel_handling='individual', target=1, limits=(None, None),
-              unit=None, return_reference=False):
+              unit=None, nan_policy='raise', return_reference=False):
```
Comment (f-brinkmann) [inline pyfar/dsp/dsp.py:2000]: I think the new parameter should go last to maintain backwards compatability

File: pyfar/dsp/dsp.py [hunk 2]
```diff
@@ -2183,9 +2206,13 @@ def normalize(signal, reference_method='max', domain='time',
             input_data = np.abs(signal.freq)
     # get values for normalization max or mean
         if reference_method == 'max':
-            reference = np.max(input_data[..., limits[0]:limits[1]], axis=-1)
+            get_max = np.nanmax if nan_policy == 'omit' else np.max
+            reference = get_max(input_data[..., limits[0]:limits[1]],
+                                axis=-1)
```
Comment (f-brinkmann) [inline pyfar/dsp/dsp.py:2211]: Why not directly use `np.nanmax(..., nan_policy=nan_policy)`?
Might also be changed in the elif case below.

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Looks good to me. I had only one minor comment and question.

**Review by f-brinkmann — APPROVED** [review]

**Review by sikersten — CHANGES_REQUESTED** [review]
Thanks for implementing!
I tested with a Signal object, `s = pf.Signal([2, -1, 0, np.nan], 44100)`. The mode `nan_policy='omit' ` does not seem to work in this case. Could you please add a test and check what the reason is?

**Comment by twennemann** [comment]
> Thanks for implementing! I tested with a Signal object, `s = pf.Signal([2, -1, 0, np.nan], 44100)`. The mode `nan_policy='omit' ` does not seem to work in this case. Could you please add a test and check what the reason is?

To apply normalization we calculate the normalized signal as followed:
`normalized_signal = signal.copy() * target / reference_norm`

If the signal object, which is used for a calculation, is type `pf.Signal` and it includes NaN values, every value will be returned with NaN.  So this problem needs to be fixed in the Signal class.

**Review by sikersten — APPROVED** [review]

**Pull request merged**